File: spring-web-java/src/main/java/spring/web/java/global/converter/persistence/CryptoAttributeConverter.java
package spring.web.java.global.converter.persistence;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;
import spring.web.java.global.common.AESCryptoUtils;

@Converter
public class CryptoAttributeConverter implements AttributeConverter<String, String> {

	private final AESCryptoUtils aesCryptoUtils;

	public CryptoAttributeConverter() {
		this.aesCryptoUtils = new AESCryptoUtils("d9ANIqIyfTygI92m6jWFfAzUbEP73TNB");
	}

	@Override
	public String convertToDatabaseColumn(String s) {
		return aesCryptoUtils.encrypt(s);
	}

	@Override
	public String convertToEntityAttribute(String s) {
		return aesCryptoUtils.decrypt(s);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/global/converter/persistence/CryptoAttributeConverter.kt
package spring.web.kotlin.global.converter.persistence

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter
import spring.web.kotlin.global.common.AESCryptoUtils

@Converter
class CryptoAttributeConverter : AttributeConverter<String?, String?> {

    override fun convertToDatabaseColumn(attribute: String?): String? {
        return attribute?.let { AESCryptoUtils.encrypt(it) }
    }

    override fun convertToEntityAttribute(dbData: String?): String? {
        return dbData?.let { AESCryptoUtils.decrypt(it) }
    }
}
